core/common-ui/src/main/java/team/noweekend/core/common/ui/calendar/CalendarDataProvider.kt

```diff
@@ -0,0 +1,268 @@
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.runtime.toMutableStateList
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import team.noweekend.core.common.ui.calendar.model.DateOfWeek
+import team.noweekend.core.common.ui.calendar.model.WeeksData
+import team.noweekend.core.common.ui.calendar.state.CalendarPagerState.ImageType
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.temporal.TemporalAdjusters
+
+@Composable
+fun rememberCalendarDataProvider(
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+): CalendarDataProvider {
+    return remember {
+        CalendarDataProvider(
+            coroutineScope = coroutineScope,
+        )
+    }
+}
+
+@Stable
+class CalendarDataProvider(
+    private val coroutineScope: CoroutineScope,
+) {
+
+    private val _targetDate: MutableState<LocalDate> = mutableStateOf(LocalDate.now())
+
+    val targetDate: State<LocalDate> = _targetDate
+
+    private val TAG: String = "CalendarDataProvider"
+
+    private val _calendarDataProviderEventChannel: Channel<CalendarDataProviderEvent> =
+        Channel(capacity = Channel.BUFFERED)
+
+    val calendarDataProviderEventFlow: Flow<CalendarDataProviderEvent> =
+        _calendarDataProviderEventChannel.receiveAsFlow()
+
+    private val currentWeekMonday: LocalDate = _targetDate.value.with(
+        TemporalAdjusters.previousOrSame(
+            DayOfWeek.MONDAY,
+        ),
+    )
+
+    private val currentMonthStart: LocalDate = _targetDate.value.with(
+        TemporalAdjusters.firstDayOfMonth(),
+    )
+
+    val weeksData: SnapshotStateList<WeeksData> = mutableListOf(
+        getWeekDates(startedMonday = currentWeekMonday.minusWeeks(1)),
+        getWeekDates(startedMonday = currentWeekMonday),
+        getWeekDates(startedMonday = currentWeekMonday.plusWeeks(1)),
+    ).toMutableStateList()
+
+    val monthData: SnapshotStateList<WeeksData> = mutableListOf(
+        getMonthDates(monthStart = currentMonthStart.minusMonths(1)),
+        getMonthDates(monthStart = currentMonthStart),
+        getMonthDates(monthStart = currentMonthStart.plusMonths(1)),
+    ).toMutableStateList()
+
+    /**
+     * 주 데이터 를 반환
+     */
+    private fun getWeekDates(startedMonday: LocalDate): WeeksData {
+        return WeeksData(
+            year = startedMonday.year,
+            month = startedMonday.monthValue,
+            dateOfWeeks = (0..6).map { day: Int ->
+                val localDate: LocalDate = startedMonday.plusDays(day.toLong())
+                DateOfWeek(
+                    imageType = ImageType.entries.random(), // Todo 이미지 로직
+                    localDate = localDate,
+                )
+            }.chunked(7).map { weeks: List<DateOfWeek> -> weeks.toImmutableList() }
```

Comment:
(0..6) 범위가 정해져서 chunked 함수 안써도 되는 것 같네?
```suggestion
            }.toImmutableList()
```

Reply:
@jaeryo2357 
요게 월별 캘린더와 주별 캘린더의 데이터 모델을 같이 쓰려고 작성했엉

```
data class WeeksData(
    val year: Int,
    val month: Int,
    /**
     * 월과 주의 데이터 관리를 위해 2차원 List 로 관리
     */
    val dateOfWeeks: ImmutableList<ImmutableList<DateOfWeek>>,
)
```
주별 캘린더의 경우 페이지에 제공되는 데이터가 
2025 6 
[
DateOfWeek(imageType=NONE, localDate=2025-06-09),
DateOfWeek(imageType=NONE, localDate=2025-06-10),
DateOfWeek(imageType=Rest, localDate=2025-06-11), 
DateOfWeek(imageType=OverFiftyUnderSeventyFive, localDate=2025-06-12), DateOfWeek(imageType=OverZeroUnderFiftyDegree, localDate=2025-06-13), DateOfWeek(imageType=FutureSchedule, localDate=2025-06-14), 
DateOfWeek(imageType=BurnOut, localDate=2025-06-15)
]

이렇게 주에 7일이 포함되니까 7개만 들어가 있는게 맞앙 

근데 월별 캘린더의 경우에 페이지에 제공되는 데이터는 (주 데이터 * 해당월이 가지고 있는 주의 수)를 가지니까
2025 5
[ 첫번째 주
DateOfWeek(imageType=OverZeroUnderFiftyDegree, localDate=2025-04-28), DateOfWeek(imageType=FutureSchedule, localDate=2025-04-29),
DateOfWeek(imageType=FutureSchedule, localDate=2025-04-30),
DateOfWeek(imageType=FutureSchedule, localDate=2025-05-01),
DateOfWeek(imageType=NONE, localDate=2025-05-02),
DateOfWeek(imageType=Rest, localDate=2025-05-03),
DateOfWeek(imageType=NONE, localDate=2025-05-04)
],
[ 두번째 주
DateOfWeek(imageType=BurnOut, localDate=2025-05-05),
DateOfWeek(imageType=BurnOut, localDate=2025-05-06),
DateOfWeek(imageType=NONE, localDate=2025-05-07),
DateOfWeek(imageType=BurnOut, localDate=2025-05-08),
DateOfWeek(imageType=Rest, localDate=2025-05-09),
DateOfWeek(imageType=Rest, localDate=2025-05-10),
DateOfWeek(imageType=OverZeroUnderFiftyDegree, localDate=2025-05-11)
],
...

List<List<DayOfWeek>>의 구조를 가져서 
같은 데이터 모델을 공유하게 하려구 chuck를 써서 List로 한번 더 랩핑하게 했엉

Reply:
아하 데이터 타입을 맞출려고 한거구나 이해 완료